toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -274,6 +278,20 @@ func PopulateInstallRoot(installChroot *safechroot.Chroot, packagesToInstall []s
 		return
 	}
 
+	if !config.RemoveRpmDb {
+		// User wants to avoid removing the RPM database.
+		logger.Log.Debug("RemoveRpmDb is not turned on. Skipping RPM database cleanup.")
+	} else {
+		defer func() {
+			// Signal an error if cleanup fails; don't overwrite the previous error though.
+			// Failure to clean up the RPM database constitutes a build break.
+			cleanupErr := cleanupRpmDatabase(installRoot)
```

Comment:
Suggestion: we could still print an error message when `err != nil` to inform about any clean-up issues.